src/systemTest/java/com/flightstats/hub/system/service/ChannelService.java

```diff
@@ -71,6 +85,77 @@ public String addItem(String channelName, Object data) {
         return response.body().get_links().getSelf().getHref();
     }
 
+    private List<String> getPathParts(String path) {
+        String itemPath = path.replace(hubBaseUrl.toString() + "channel/", "");
+        return Arrays.asList(itemPath.split("/"));
+    }
+
+    private Map<DatePathIndex, Integer> getPathDateParts(List<String> pathParts) {
+        Map<DatePathIndex, Integer> dateValues = new HashMap<>();
+        List<Integer> dateParts = pathParts.subList(1, pathParts.size() -1).stream()
+                .filter(StringUtils::isNotBlank)
+                .map(Integer::parseInt)
+                .collect(Collectors.toList());
+        dateValues.put(DatePathIndex.YEAR, dateParts.get(DatePathIndex.YEAR.getIndex()));
+        dateValues.put(DatePathIndex.MONTH, dateParts.get(DatePathIndex.MONTH.getIndex()));
+        dateValues.put(DatePathIndex.DAY, dateParts.get(DatePathIndex.DAY.getIndex()));
+        dateValues.put(DatePathIndex.HOUR, dateParts.get(DatePathIndex.HOUR.getIndex()));
+        dateValues.put(DatePathIndex.MINUTE, dateParts.get(DatePathIndex.MINUTE.getIndex()));
+        dateValues.put(DatePathIndex.SECONDS, dateParts.get(DatePathIndex.SECONDS.getIndex()));
+        dateValues.put(DatePathIndex.MILLIS, dateParts.get(DatePathIndex.MILLIS.getIndex()));
+        return dateValues;
+    }
+
+    private Map<String, String> getPathKeys(List<String> pathParts) {
+        Map<String, String> keys = new HashMap<>();
+        keys.put("channelName", pathParts.get(0));
+        keys.put("key", pathParts.get(pathParts.size() - 1));
+        return keys;
+    }
+
+    @SneakyThrows
+    private Optional<TimeQuery> getItemByTimeFromLocation(String path, Location location) {
+        List<String> pathParts = getPathParts(path);
+        Map<String, String> keys = getPathKeys(pathParts);
+        Map<DatePathIndex, Integer> dateParts = getPathDateParts(pathParts);
+        Call<TimeQuery> response = channelItemResourceClient.getItemsSecondsPath(keys.get("channelName"),
+                dateParts.get(DatePathIndex.YEAR),
```

Comment:
It sort of feels like there might be value in an actual DatePath class.

Reply:
True